src/Connections/SetupConnection.ts

```diff
@@ -161,42 +163,54 @@ export class SetupConnection extends CommandConnection {
         return this.as.botClient.sendHtmlNotice(this.roomId, md.renderInline(`Room configured to bridge Figma file.`));
     }
 
-    @botCommand("feed", { help: "Bridge an RSS/Atom feed to the room.", requiredArgs: ["url"], includeUserId: true, category: "feed"})
-    public async onFeed(userId: string, url: string) {
+    @botCommand("feed", { help: "Bridge an RSS/Atom feed to the room.", requiredArgs: ["url"], optionalArgs: ["label"], includeUserId: true, category: "feed"})
+    public async onFeed(userId: string, url: string, label?: string) {
         if (!this.config.feeds?.enabled) {
             throw new CommandError("not-configured", "The bridge is not configured to support feeds.");
         }
 
         await this.checkUserPermissions(userId, "feed", FeedConnection.CanonicalEventType);
 
+        // provisionConnection will check it again, but won't give us a nice CommandError on failure
         try {
-            new URL(url);
-            // TODO: fetch and check content-type?
-        } catch {
+            await FeedConnection.validateUrl(url);
+        } catch (err: unknown) {
+            log.error(`Feed URL '${url}' failed validation: ${err}`);
```

Comment:
Ultra nit: Do we want to log these as error? I'm in the school of thought that `error` should be reserved for "there is something wrong with the application / config" rather than the user maybe mistyping a typo.

Reply:
Yeah, it's probably not the right choice. My rationale was that, since this error won't actually appear on Matrix anywhere it should be easy to find (or at least be logged no matter the loglevel) in the logfiles. That said, `debug` is probably more fitting (and agreed that `error` should be reserved for errors in SetupConnection, in this context.